tests/test_lowlevel.py

```diff
@@ -204,3 +203,26 @@ def test_use_proxy_from_environment(httpbin, var, scheme):
 
         # it had actual content (not checking for SOCKS protocol for now)
         assert len(fake_proxy.handler_results[0]) > 0
+
+
+def test_redirect_rfc1808_to_non_ascii_location():
+    def redirect_resp_handler(sock):
+        consume_socket_content(sock, timeout=0.5)
+        sock.send(
+            b'HTTP/1.1 301 Moved Permanently\r\n'
+            b'Location: //%s:%d/?\xc5\xa0\r\n'
+            b'\r\n' % (host.encode(), port)
+        )
+        consume_socket_content(sock, timeout=0.5)
+        sock.send(b'HTTP/1.1 200 OK\r\n\r\n')
+
+    close_server = threading.Event()
+    server = Server(redirect_resp_handler, wait_to_close_event=close_server)
+
+    with server as (host, port):
+        url = u'http://{0}:{1}'.format(host, port)
+        r = requests.head(url=url, allow_redirects=True)
+        assert r.status_code == 200
+        assert len(r.history) == 1
+        assert r.history[0].status_code == 301
```

Comment:
add the assertion for url.
please let me know if there is a better way or more common way to ignore this assertion for python3.

please let me know if any other changes needed :smile: